#4 feature(lab4): Реализовал 4 лабораторную работу

inittose opened the pull request:
- Добавил класс PriorityOrder и перечисление OrderTime
- Реализовал логику приоритетных покупателей и их заказов
- Обновил UML-диаграмму

inittose committed:
feature(lab3): Создал класс Cart

inittose committed:
feature(lab3): Добавил класс Order

- Создал перечисление OrderStatus

inittose committed:
feature(lab3): Сверстал вкладку CartsTab

inittose committed:
feature(lab3): Реализовал локигу вкладки CartsTab

inittose committed:
feature(lab3): Сверстал OrdersTab

- DataGridView еще не привязана к данным

inittose committed:
feature(lab3): Реализовал логику вкладки OrdersTab

- Добавил сериализацию Id и DateTime в Order
- Добавил список занятый Id, теперь Id всегда статичны и не повторяются
- Попровавил некоторые комментарии

inittose committed:
docs(lab3): UML-диаграмма для 3 лабы

inittose committed:
style(lab3): Поправил верстку во вкладках

- Также поправил UML-диаграмму
- for заменил на foreach

inittose committed:
style(lab3): больше неявной типизации и foreach

inittose committed:
feature(lab4): Реализовал класс PriorityOrder

- Создал перечисление OrderTime

inittose committed:
feature(lab4): Добавил верстку PriorityOrdersTab

inittose committed:
feature(lab4): Реализовал логику работы Priority

- Добавил Priority для Customer
- Добавил настройку для сериализатора TypeNameHandling

inittose committed:
docs(lab4): Обновил UML-диаграмму

- Добавил пометку для Priority Order

inittose committed:
style(lab4): Добавил комментарии

inittose committed:
docs(lab4): Обновил UML-диаграмму в README

Nikita-Isaichenko commented on ObjectOrientedPractics/ObjectOrientedPractics/Model/Cart.cs:
```diff
@@ -0,0 +1,27 @@
+    public class Cart
+    {
+        /// <summary>
+        /// Список товаров <see cref="Item"/> в корзине.
```
Свойства комментируй единообразно, добавь "Возвращает и задает...". Проверь во всем коде

Nikita-Isaichenko commented on ObjectOrientedPractics/ObjectOrientedPractics/View/Tabs/CartsTab.cs:
```diff
@@ -0,0 +1,246 @@
+                }
+            }
+
+            
```
Убрать лишнюю строку. "Используйте одну пустую строку между логическими частями в методе" из RSDN. Проверить везде

Nikita-Isaichenko commented on ObjectOrientedPractics/ObjectOrientedPractics/View/Tabs/CartsTab.cs:
```diff
@@ -0,0 +1,246 @@
+            if (CurrentCustomer < 0 || CartListBox.SelectedIndex < 0)
+            {
+                return;
+            }
```
Стоит поставить пустую строку. https://rsdn.org/article/mag/200401/codestyle.XML прочитай здесь когда ставится пустая строка, а когда нет, проверь по всему коду

inittose committed:
style(lab4): Расставил пустые строки

- Назвал все свойства по шаблону
- Реализовал логику удаления данных при ошибочной десериализации

CephalonGL commented on ObjectOrientedPractics/ObjectOrientedPractics/Services/ItemFactory.cs:
```diff
@@ -35,8 +34,7 @@ public static Item GetRandomItem()
             var itemInfo = randomData[2];
             var randomCategory = (Category)random.Next(0, Enum.GetNames(typeof(Category)).Length);
```
Из-за того, то в одной строке делаешь много действий одновременно, падает читаемость кода. Когда это будет читать другой разработчик или ты через пару недель, будет труднее в этом разобраться. Поэтому всегда, когда пишешь код, старайся его оформлять его так, чтобы его можно было легче понять читающему. Расписать в несколько строк

CephalonGL commented on ObjectOrientedPractics/ObjectOrientedPractics/View/MainForm.cs:
```diff
@@ -35,5 +39,33 @@ private void MainForm_FormClosed(object sender, FormClosedEventArgs e)
+        {
+            switch (MainTabControl.SelectedIndex)
+            {
+                case 2:
```
Содержимое кейсов обернуть в фигурные скобки

CephalonGL commented on ObjectOrientedPractics/ObjectOrientedPractics/Services/Serializer.cs:
```diff
@@ -50,7 +51,21 @@ public static Store GetStore()
+                catch
+                {
+                    StoreJson = string.Empty;
+                    MessageBox.Show("Данные повреждены.\nФайлы сохранения очищены.");
```
Здесь на русском, в остальных местах программы - на английском. Сделать единообразно
